macros/src/interface/parse.rs

```diff
@@ -85,6 +85,44 @@ pub mod cooked {
 					raw::InterfaceItem::Stream(raw) => streams.push(StreamDefinition::from_raw(errors, raw)),
 				}
 			}
+
+			// Remove things with duplicate names, because they'll cause a lot more compile errors.
+			// Duplicate IDs we just generate though, because they don't cause duplicate type or functions name.
+			let mut remove_services = Vec::new();
+			let mut remove_streams = Vec::new();
+
+			for (a_i, a) in services.iter().enumerate() {
+				for (b_i, b) in services.iter().enumerate().skip(a_i + 1) {
+					if a.service_id.value == b.service_id.value {
+						errors.push(syn::Error::new(b.service_id.span, "duplicate service ID"));
+					}
+					if a.name() == b.name() {
+						errors.push(syn::Error::new(b.name().span(), "duplicate service name"));
+						remove_services.push(b_i);
+					}
+				}
+			}
+
+			for (a_i, a) in streams.iter().enumerate() {
+				for (b_i, b) in streams.iter().enumerate().skip(a_i + 1) {
+					if a.service_id.value == b.service_id.value {
+						errors.push(syn::Error::new(b.service_id.span, "duplicate service ID"));
+					}
+					if a.name() == b.name() {
+						errors.push(syn::Error::new(b.name().span(), "duplicate stream name"));
+						remove_streams.push(b_i);
+					}
+				}
+			}
+
+			for i in remove_services.into_iter().rev() {
+				services.remove(i);
+			}
```

Comment:
Looks like a mere duplicate :P

Reply:
It does the same thing, but the types are different. So it can't really be de-duplicated that easily.